lib/Parse/ParseType.cpp

```diff
@@ -122,6 +132,11 @@ ParserResult<TypeRepr> Parser::parseTypeSimple(Diag<> MessageID,
         ty = parseTypeImplicitlyUnwrappedOptional(ty.get());
         continue;
       }
+      // Parse legacy array types for migration.
```

Comment:
I'm a little concerned about someone has *bad* habit typing `Int[]`, then `command` + `'` on Xcode.
As far as there's no ambiguity here, I think we should keep this here. Any thought?